ethereum/swap_factory_test.go

```diff
@@ -44,18 +46,71 @@ func testNewSwap(t *testing.T, asset ethcommon.Address) {
 	require.NotEqual(t, ethcommon.Address{}, address)
 	require.NotNil(t, tx)
 	require.NotNil(t, contract)
+
 	receipt, err := block.WaitForReceipt(context.Background(), conn, tx.Hash())
 	require.NoError(t, err)
 	t.Logf("gas cost to deploy SwapFactory.sol: %d", receipt.GasUsed)
 
-	nonce := big.NewInt(0)
-	tx, err = contract.NewSwap(auth, [32]byte{}, [32]byte{},
-		ethcommon.Address{}, defaultTimeoutDuration, asset, big.NewInt(0), nonce)
+	owner := auth.From
+	claimer := common.EthereumPrivateKeyToAddress(tests.GetMakerTestKey(t))
+
+	var pubKeyClaim, pubKeyRefund [32]byte
+	_, err = rand.Read(pubKeyClaim[:])
 	require.NoError(t, err)
-	receipt, err = block.WaitForReceipt(context.Background(), conn, tx.Hash())
+	_, err = rand.Read(pubKeyRefund[:])
+	require.NoError(t, err)
+
+	nonce, err := rand.Prime(rand.Reader, 256)
+	require.NoError(t, err)
+	value, err := rand.Prime(rand.Reader, 53) // (2^54 - 1) / 10^18 ~= .018 ETH
+	require.NoError(t, err)
+
+	isEthAsset := asset == ethAssetAddress
+
+	if isEthAsset {
+		auth.Value = value
+	} else {
+		value = big.NewInt(0)
+	}
+
+	tx, err = contract.NewSwap(auth, pubKeyClaim, pubKeyRefund,
+		claimer, defaultTimeoutDuration, asset, value, nonce)
 	require.NoError(t, err)
 
+	receipt, err = block.WaitForReceipt(context.Background(), conn, tx.Hash())
+	require.NoError(t, err)
 	t.Logf("gas cost to call new_swap: %d", receipt.GasUsed)
+
+	newSwapLogIndex := 0
+	if !isEthAsset {
+		newSwapLogIndex = 2
+	}
+	require.Equal(t, newSwapLogIndex+1, len(receipt.Logs))
+
+	swapID, err := GetIDFromLog(receipt.Logs[newSwapLogIndex])
+	require.NoError(t, err)
+
+	t0, t1, err := GetTimeoutsFromLog(receipt.Logs[newSwapLogIndex])
+	require.NoError(t, err)
+
+	// validate that off-chain swapID calculation matches the on-chain value
+	swap := SwapFactorySwap{
+		Owner:        owner,
+		Claimer:      claimer,
+		PubKeyClaim:  pubKeyClaim,
+		PubKeyRefund: pubKeyRefund,
+		Timeout0:     t0,
+		Timeout1:     t1,
+		Asset:        ethcommon.Address(types.EthAssetETH),
+		Value:        value,
+		Nonce:        nonce,
+	}
+
+	// validate our off-net calculation of the SwapID
+	if !isEthAsset {
+		t.Skip("off-net calculation of swapID doesn't match")
+	}
```

Comment:
I'll have to look at this again tomorrow with fresh eyes. Any idea what field in `swap` isn't matching up to generate the correct hash when this test is running with ERC20 tokens?